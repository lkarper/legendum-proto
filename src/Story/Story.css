.Story__container {
    position: relative;
    display: flex;
    flex-direction: column;
    align-items: center;
    width: 100%;
    height: calc(100vh - 75px);
}

.Story__h2 {
    height: 50px;
    margin: 5px;
    z-index: 10;
}