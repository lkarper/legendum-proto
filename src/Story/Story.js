import React, { useState, useEffect } from 'react';
import DialoguePage from '../DialoguePage/DialoguePage';
import config from '../config';
import './Story.css';

const Story = (props) => {

    const { chapt } = props.match.params;

    const [dialogue, setDialogue] = useState([]);
    const [page, setPage] = useState(1);
    const [response, setResponse] = useState();    

    useEffect(() => {
        fetch(`${config.API_ENDPOINT}/dialogue/${chapt}`)
            .then(res => {
                if (res.ok) {
                    return res.json();
                }
                throw new Error (res.statusText);
            })
            .then(data => {
                data.sort((a, b) => a.page - b.page);
                setDialogue(data);
            })
            .catch(error => {
                console.log('error', error);
            });
    }, [chapt]);

    return (
        <div className='Story__container'>
            {dialogue.length 
                ? 
                    <>
                        <h2 className='Story__h2'>{dialogue[0].story_title}</h2>
                        <DialoguePage 
                            data={{
                                response,
                                page,
                                setPage,
                                setResponse,
                                dialogue,
                                chapt,
                            }}
                        />
                    </>
                :
                    <>
                        <p className='Story__loading'>Loading...</p>
                    </>
            }
        </div>
    );
}

export default Story;