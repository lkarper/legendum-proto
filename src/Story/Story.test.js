import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Story from './Story';

describe('Story component', () => {
    it('renders the UI without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Story />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(<Story />);
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});
