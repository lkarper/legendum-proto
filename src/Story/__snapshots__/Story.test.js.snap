// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Story component renders the UI as expected 1`] = `
<Story
  match={
    Object {
      "params": Object {
        "chapt": "",
      },
    }
  }
>
  <div
    className="Story__container"
  >
    <LoadingSpinner>
      <div
        className="LoadingSpinner__loading-container"
      >
        <FontAwesomeIcon
          aria-label="Loading."
          className="LoadingSpinner__spinner"
          icon={
            Array [
              "fa",
              "spinner",
            ]
          }
          spin={true}
        >
          <i
            className="fa"
          />
        </FontAwesomeIcon>
      </div>
    </LoadingSpinner>
    <div
      className="Story__alert-div"
      role="alert"
    />
  </div>
</Story>
`;
